mod.sp

```diff
@@ -7,6 +7,57 @@ locals {
   }
 }
 
+variable "common_dimensions" {
+  type        = list(string)
+  description = "A list of common dimensions to add to each control."
+  # Define which common dimensions should be added to each control.
+  # - connection_name (_ctx ->> 'connection_name')
+  # - region
+  # - tenant_id
+  default     = ["region", "tenant_id"]
+}
+
+variable "tag_dimensions" {
+  type        = list(string)
+  description = "A list of tags to add as dimensions to each control."
+  # A list of tag names to include as dimensions for resources that support
+  # tags (e.g.  "Department", "Environment"). Default to empty since tag names are
+  # a personal choice.
+  default     = []
+}
+
+locals {
+
+  # Local internal variable to build the SQL select clause for common
+  # dimensions using a table name qualifier if required. Do not edit directly.
+  common_dimensions_qualifier_sql = <<-EOQ
+  %{~ if contains(var.common_dimensions, "connection_name") }, __QUALIFIER___ctx ->> 'connection_name' as connection_name%{ endif ~}
+  %{~ if contains(var.common_dimensions, "region") }, __QUALIFIER__region as region%{ endif ~}
+  %{~ if contains(var.common_dimensions, "tenant_id") }, __QUALIFIER__tenant_id as tenant_id%{ endif ~}
+  EOQ
+
+  common_dimensions_qualifier_global_sql = <<-EOQ
+  %{~ if contains(var.common_dimensions, "connection_name") }, __QUALIFIER___ctx ->> 'connection_name' as connection_name%{ endif ~}
+  %{~ if contains(var.common_dimensions, "tenant_id") }, __QUALIFIER__tenant_id as tenant_id%{ endif ~}
+  EOQ
+
+  # Local internal variable to build the SQL select clause for tag
+  # dimensions. Do not edit directly.
+  tag_dimensions_qualifier_sql = <<-EOQ
+  %{~ for dim in var.tag_dimensions },  __QUALIFIER__tags ->> '${dim}' as "${replace(dim, "\"", "\"\"")}"%{ endfor ~}
+  EOQ
+
```

Comment:
@khushboo9024 as discussed, we should check the compartment name variable as well.